src/main/java/com/zufar/onlinestore/payment/converter/PaymentIntentConverter.java

```diff
@@ -5,29 +5,39 @@
 import com.zufar.onlinestore.payment.calculator.PaymentPriceCalculator;
 import com.zufar.onlinestore.payment.dto.CreatePaymentDto;
 import com.zufar.onlinestore.payment.entity.Payment;
-import lombok.RequiredArgsConstructor;
-import org.springframework.stereotype.Component;
+import org.mapstruct.Mapper;
 
-@RequiredArgsConstructor
-@Component
-public class PaymentIntentConverter {
+@Mapper
+public interface PaymentIntentConverter {
 
-    private final PaymentPriceCalculator paymentPriceCalculator;
+    PaymentPriceCalculator paymentPriceCalculator = new PaymentPriceCalculator();
 
-    public Payment toPayment(final PaymentIntent paymentIntent) {
-        return Payment.builder()
-                .itemsTotalPrice(paymentPriceCalculator.calculatePriceForPayment(paymentIntent.getAmount()))
-                .paymentIntentId(paymentIntent.getId())
-                .currency(paymentIntent.getCurrency())
-                .build();
+    default Payment toPayment(final PaymentIntent paymentIntent) {
+        if (paymentIntent == null) {
+            return null;
+        }
+
+        Payment.PaymentBuilder builder = Payment.builder();
+
+        builder.itemsTotalPrice(paymentPriceCalculator.calculatePriceForPayment(paymentIntent.getAmount()));
+        builder.paymentIntentId(paymentIntent.getId());
+        builder.currency(paymentIntent.getCurrency());
+
+        return builder.build();
     }
 
-    public PaymentIntentCreateParams toPaymentIntentParams(final CreatePaymentDto createPaymentDto) {
-        return PaymentIntentCreateParams.builder()
-                .setAmount(paymentPriceCalculator.calculatePriceForPaymentIntent(
-                        createPaymentDto.priceDetails().totalPrice()))
-                .setCurrency( createPaymentDto.priceDetails().currency())
-                .setPaymentMethod(createPaymentDto.paymentMethodId())
-                .build();
+    default PaymentIntentCreateParams toPaymentIntentParams(final CreatePaymentDto createPaymentDto) {
+        if (createPaymentDto == null) {
+            return null;
+        }
+
+        PaymentIntentCreateParams.Builder builder = PaymentIntentCreateParams.builder();
```

Comment:
Do we really need Builders for Mapstruct? 
I sow that Mapstruct provides set of annotations which we can use in our app to avoid boilerplate code